app/lib/shared/analyzer_client.dart

```diff
@@ -33,6 +34,7 @@ final Logger _logger = new Logger('pub.analyzer_client');
 /// Client methods that access the analyzer service and the internals of the
 /// analysis data. This keeps the interface narrow over the raw analysis data.
 class AnalyzerClient {
+  final Map<AnalysisKey, AnalysisExtract> _extractCache = {};
```

Comment:
How do we make sure this doesn't grow unbounded?

Reply:
good point, added a max limit